### PR TITLE
Frontend  connect frontend and backend (call backend ap is)

### PR DESCRIPTION
**Pull request by Andrew-Ham**
**This is a poorly named branch I apologize for that. This branch is just solely about frontend firebase authentication** - This branch I want to merge into Master is where I completed the frontend authentication. Read below on the technical functionalities I have met.

Authentication is complete on frontend side. User can register with email and password. This information about user is populated in our Firebase database.

User can also log in with their signed-up accounts. UI changes based on the login status.

The **next requirement** to work on is to connect with the backend. As discussed in meetings I must send the Firebase token to our backend APIs when a user signs up for an event and goes to payment.

### CODE REVIEW

File: packages/frontend/src/routes/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -14,25 +19,34 @@ export default function SignUp() {
...
+    setPasswordError(false);
+    setLoading(true);
+
+    try {
+      await createUserWithEmailAndPassword(auth, email, password).then(
+        (userCredential) => {
+          const user = userCredential.user;
+          console.log("User SIGNED UP ith the email", auth?.currentUser?.email);
+          setCurrentUser(user);
+          navigate("/");
```
Comment (JC4github) [inline packages/frontend/src/routes/SignUp/SignUp.jsx:42]: can we navigate to /login after signup, or does it automatically log user in after sign up?

**Review by JC4github — COMMENTED** [review]
Just a question redirecting the user after signup, is it possible to navigate user to login route and not the home route?

File: packages/frontend/src/routes/SignUp/SignUp.jsx (hunk 1 above)
Comment (Andrew-Ham) [reply packages/frontend/src/routes/SignUp/SignUp.jsx:42]: Yea it is possible to redirect the user to login page after signup and make them log in again. Would just need to not set the user state and set logged in to false.
But isn't it better if a user automatically signs in after signing up? Would be inconvenient for them to log in again.

Comment (JC4github) [reply packages/frontend/src/routes/SignUp/SignUp.jsx:42]: oh nice, so if it can auto login user after signup then that's better

**Review by JC4github — APPROVED** [review]
No problems. Great job!

**Pull request merged**